### DISCUSSION TITLE
Feature: Use shared UDP port

### DISCUSSION THREAD

**Discussion by asoMansoury** [vpnhood/VpnHood#326] [General]
Can you describe what is this new feature and how do i can update my previous VpnHood server to the new version you have released?

Can you explain in which way I can enable this new feature on Console?

Is this feature to specify unique UDP port on VpnHood server like 443 TCP Port?

**Comment by trudyhood** [comment]
Yes, it is still in beta, and it would be great if you could check it out. I updated the document of UDP so you can change the port
https://github.com/vpnhood/VpnHood/wiki/VpnHood-Server-Configuration

If you are using the access server, you need to wait a few days as the team works on it.

**Reply by asoMansoury** [reply]
It's great to hear it from you.
I'm really waiting for setting a unique port for UDP, my clients in Iran really waiting for such useful feature.
As I know you are not so aware about status of filtering in Iran, I can say you, your Vpn is the only Vpn that works fine in Iran specially in compare of V2ray.

**Comment by trudyhood** [comment]
It is done. However, We have to wait about 48 hours and maybe more for google play until all servers and clients receive the updates.
[v2.9.369](https://github.com/vpnhood/VpnHood/releases/tag/v2.9.369)

**Comment by asoMansoury** [comment]
I'm Impatiently waiting for using this great feature.
I read the documentation about UDP Port , and I found that the default port for UDP is 1024, can I change it to another port or it is static?
Is is changeable like of 443 of TCP Via console.vpnhood.com or not, and enable listening option?

**Comment by trudyhood** [comment]
The default is not 1024! It is a random port assigned to your server. You can change it in Control Panel
![image](https://github.com/vpnhood/VpnHood/assets/72891124/e8e65c2d-127a-4249-a537-565cceb14b10)

**Comment by asoMansoury** [comment]
Thanks a lot.
I enabled the UPD port on my old
 servers and it did not work, what I should do to enable this feature on previous servers?
How do I can update my old servers?

![tunnel](https://github.com/vpnhood/VpnHood/assets/14541559/c9b2db2f-f465-4da8-8a38-e4f4449dbae3)
I enabled Tunneling on my own Iran's server and my clients connect to foreign server via Iran which I have enabled tunneling to specific port as you can see on the above pic.

**Reply by trudyhood** [reply]
Your server will be updated automatically in one day. Check the access server, and it should be 2.9.370. Clients must be 2.9.369
![image](https://github.com/vpnhood/VpnHood/assets/72891124/5be05c40-1b16-464e-8206-65d741219c74)


Try to install it manually or run vhupdate from the VpnHoodServer location if it is not updated.

**Comment by trudyhood** [comment]
Make sure it is 2.9.370.
> I enabled the UPD port on my old servers and it did not work

How do you check it?

**Comment by asoMansoury** [comment]
Yes I see all server are updated to the latest version, also I installed the latest windows client version, but still it does not work on UDP when I tunnel from iran.

**Comment by trudyhood** [comment]
send me an access key so i can check. Are you sure UDP is not blocked in your country?

**Reply by asoMansoury** [reply]
Yes I'm sure it is not blocked in my country.
Is Tunneled all my Udp traffic to main server by this command on ubuntu
iptables -t nat -A PREROUTING -p UDP  -j DNAT --to-destination MainServer;

**Reply by trudyhood** [reply]
Thank you, I will check it tomorrow

**Comment by trudyhood** [comment]
To ensure the security of your access key and prevent the sharing of your public IP address, please refrain from sending any sensitive information, such as access keys, via email or any other unsecured means.

Regarding the issue you mentioned, I can assist you with checking the functionality of the VpnHood Server, specifically with regard to serving UDP with a shared IP address. However, please note that I won't be able to inspect the forwarder directly. If the VpnHood Server is functioning properly, the issue may lie with your forwarder, and further troubleshooting will be required to identify and resolve the problem.

**Reply by asoMansoury** [reply]
Hello for debugging and to see that the packets are forwarding from the tunneled server to the Main server ,I can give you the user and password, to see the status of UDP packets and check that is the packets are going through the tunneled server to the main server.
If you are Ok with it,let me to know to send you the username and password.

tcpdump is a great tools in ubuntu to monitoring all packets.
For example by using the below command you can check what is happening with UDP packets.
tcpdump -n udp port {port number}

it is an example wich I'm sure about UDP Packet which are sent to the main server.
21:34:59.870630 IP tunneledServer:45676 > MainServer.1029: UDP, length 0

**Comment by trudyhood** [comment]
Our UDP tunneling appears to be functioning correctly. However, we currently lack the time to diagnose the port-forwarding feature. It is possible that either your server is not forwarding UDP packets correctly or that your network infrastructure is filtering certain UDP packets. It should be noted that UDP protocol reliability varies in different countries, which is why the default protocol in Vpnhood is "no UDP." Your assistance in identifying the specific problem would be greatly appreciated. Rest assured, if you can pinpoint the underlying issue, we will take responsibility for resolving it.
